kernel/src/types.ts

```diff
@@ -17,8 +19,17 @@ export interface Interaction {
   outputs: InteractionOutput[];
 }
 
+/* Interactions */
+
 export interface InteractionInput {
-  text: string;
+  text?: string;
```

Comment:
Made text optional, needs more testing. I assume you need to be able to just send a file to the model without adding any additional context.